tests/countries/test_cocos_islands.py

```diff
@@ -0,0 +1,291 @@
+
+    def test_act_of_self_determination_day(self):
+        name = "Act of Self Determination Day"
+        self.assertHolidayName(
+            name,
+            "2007-04-05",
+            (f"{year}-04-06" for year in (*range(1985, 2007), *range(2008, 2050))),
+        )
+        obs_dt = (
+            "2013-04-08",
+            "2014-04-07",
+            "2019-04-10",
+            "2024-04-08",
+            "2025-04-07",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_good_friday(self):
+        name = "Good Friday"
+        self.assertHolidayName(
+            name,
+            "2020-04-10",
+            "2021-04-02",
+            "2022-04-15",
+            "2023-04-07",
+            "2024-03-29",
+            "2025-04-18",
+        )
+        self.assertHolidayName(name, range(1985, 2050))
+
+    def test_easter_monday(self):
+        name = "Easter Monday"
+        self.assertHolidayName(
+            name,
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+        )
+        self.assertHolidayName(name, range(1985, 2050))
+
+    def test_anzac_day(self):
+        name = "ANZAC Day"
+        self.assertHolidayName(name, (f"{year}-04-25" for year in range(1985, 2050)))
+        obs_dt = (
+            "2020-04-27",
+            "2021-04-26",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_kings_birthday(self):
+        name_king = "King's Birthday"
+        name_queen = "Queen's Birthday"
+        # Queen's Birthday before 2022
+        self.assertHolidayName(
+            name_queen,
+            "2020-06-08",
+            "2021-06-07",
+            "2022-06-06",
+        )
+        # King's Birthday from 2022 onward
+        self.assertHolidayName(
+            name_king,
+            "2023-06-06",
+            "2024-06-06",
+            "2025-06-09",
+        )
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(1985, 2050)))
+        obs_dt = (
+            "2022-12-27",
+            "2033-12-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(1985, 2050)))
+
+    def test_islamic_new_year(self):
+        name = "Islamic New Year"
+        self.assertHolidayName(
+            name,
+            "2007-01-22",
+            "2008-01-10",
+            "2009-12-18",
+            "2010-12-07",
+            "2013-11-04",
+            "2014-10-25",
+            "2016-10-03",
+            "2017-09-22",
+            "2019-09-01",
+        )
```

Comment:
```suggestion
        self.assertHolidayName(
            name,
            self.no_estimated_holidays,
            "2015-10-14",
            "2016-10-03",
            "2017-09-22",
            "2018-09-11",
            "2019-09-01",
        )
        self.assertHolidayName(name, self.no_estimated_holidays, range(1985, 2020))
        self.assertNoHolidayName(name, self.no_estimated_holidays, range(2020, 2050))
        obs_dt = (
            "1999-04-19",
            "2004-02-23",
            "2011-11-28",
            "2014-10-27",
            "2019-09-02",
        )
        self.assertHolidayName(f"{name} (observed)", self.no_estimated_holidays, obs_dt)
        self.assertNoNonObservedHoliday(obs_dt)
```
And similar for the rest of Islamic holidays.